### ISSUE TITLE
task cancellation

### ISSUE THREAD

**Issue by martlaboloa** [redux-saga/redux-saga#1185]
Question:
what exactly happens when forked task has been cancelled ?
it jumps to finally block,I know that much, but
how exactly is execution halted ? should I assume that it can be stopped anywhere, like in between every statement? or does it halt only on yield statements, and if so after yield executed or before, or both can happen?

**Comment by Andarist** [comment]
Generators can be paused only "on" yield statements. The yielded effects are getting interpreted by the redux-saga runtime. When the effect resolves your generator is getting resumed with the resolved result.

So when saga cancellation happens before effect get the chance to resolve, redux-saga runtime cancels all pending effects attached to it and calls smth like `yourSaga.return()` (iterators created by calling generators have `return` method on them).

Calling `return` causes generators to jump into `finally` blocks.

To sum it up - cancellation can only happen "after" yield statement, "before" the generator resumes its work.
